R/utils.R

```diff
@@ -86,10 +86,13 @@ getMultipeptide <- function(precursors, features, applyFun=lapply, masters = NUL
       num_analytes <- length(analytes)
       newdf <- rbindlist(lapply(runs, function(run){
         df <- rbindlist(list(features[[run]][list(analytes), ],
-                  dummyTbl(analytes)), use.names=TRUE) # dummy for new features
+                  dummyTbl(analytes, runType)), use.names=TRUE) # dummy for new features
         set(df, j = c("run", "alignment_rank"), value = list(run, NA_integer_))
         df
       }), use.names=TRUE)
+      if ( runType=="DIA_IPF" ){
+        setcolorder(newdf, c('transition_group_id', 'feature_id', 'RT', 'intensity', 'leftWidth', 'rightWidth', 'peak_group_rank', 'm_score', 'run', 'alignment_rank', 'ms2_m_score'))
```

Comment:
Add `setcolorder` in line 52

Reply:
Not sure, why do we need to use `setcolorder`?

Reply:
> Add `setcolorder` in line 52

Just to make sure, which line is 52? You want me to add it to the `importFrom` statement? `#' @importFrom data.table rbindlist set setkeyv`

> Not sure, why do we need to use setcolorder?

I use `setcoloder` here to make sure the first 10 columns are `'transition_group_id', 'feature_id', 'RT', 'intensity', 'leftWidth', 'rightWidth', 'peak_group_rank', 'm_score', 'run', 'alignment_rank'`, otherwise this causes downstream issues when using the numbered indexes. If we don't `setcolorder`, then `alignment_rank` gets added to the very end (when there are other extra columns, i.e. `ms2_m_score`), and the rank updating will not be updating the right column anymore then.

Reply:
Right. It would look like `@importFrom data.table rbindlist set setkeyv setcoloder`

Reply:
Got it. Line 90 adds run and alignment_rank.